src/harness/testcases/WINNF_FT_S_PAT_testcase.py

```diff
@@ -104,25 +103,43 @@ def test_WINNF_FT_S_PAT_1(self, config_filename):
     
     config = loadConfig(config_filename)
     num_tests = len(config)
-    test_pass_threshold = 1.00
-    passed_tests = 0.0
-    for configItem in  config:
-       request= config[configItem]
-       refResponse = computePropagationAntennaModel(request)
-       sasResponse = self._sas_admin.QueryPropagationAndAntennaModel(request)
-
-       # Check response
-       this_test = False
-       if 'pathlossDb' in refResponse and 'txAntennaGainDbi' in refResponse:
-           this_test = (sasResponse['pathlossDb'] < refResponse['pathlossDb'] + 1) and (sasResponse['txAntennaGainDbi'] < (refResponse['txAntennaGainDbi'] + .2))
-               
-           if 'rxAntennaGainDbi' in refResponse:
-               this_test = this_test and (sasResponse['rxAntennaGainDbi'] < (refResponse['rxAntennaGainDbi'] + .2))
-       else:
-           self.assertEqual(sasResponse, refResponse)
-           this_test = 1.0
+    test_pass_threshold = 0.999 #percentage of passed test expected
+    num_passed_tests = 0.0
+    num_tests = len(config)
+    max_fail_num = int ((1-test_pass_threshold) * num_tests)
+    num_failed_tests = 0.0
+    num_invalid_tests = 0.0
+    for request in  config:
+        try:
+           refResponse = computePropagationAntennaModel(request)
+           sasResponse = self._sas_admin.QueryPropagationAndAntennaModel(request)
```

Comment:
I think you need this to be outside the 'try' block: otherwise an HTTP error from the SAS (which will be turned into an exception by the test harness) will be logged as an invalid config, even if the config was valid.

Really, the 'try' block should contain only line 114, as suggested in the pseudocode I wrote in a previous comment:

```
try:
  # call ref prop model
except InvalidArgumentError as e:
  # log invalid params for debugging by the tester
  # increment a "failed configs" counter so that the % of passed configs is correctly calculated at the end of the test
  # skip this config

# call the SAS UUT's prop model function (if we've made it this far, the ref model has already validated the input)
# comparison of SAS UUT, ref model

percentage_of_passed_tests = num_passed_tests / (num_total_tests - num_failed_configs)
```